rmf_site_editor/src/widgets/creation.rs

```diff
@@ -16,313 +16,656 @@
+                                            params
+                                                .descriptions
+                                                .get(description)
+                                                .map(|n| n.as_str())
+                                                .unwrap_or("")
+                                        } else {
+                                            ""
+                                        };
+
+                                    let mut selected_new_description = None;
+                                    ComboBox::from_id_source("choose_model_description")
+                                        .selected_text(selected_description_text)
+                                        .show_ui(ui, |ui| {
+                                            let Ok(children) = params.children.get(site_entity)
+                                            else {
+                                                return;
+                                            };
+                                            for child in children {
+                                                if let Ok(name) = params.descriptions.get(*child) {
+                                                    ui.selectable_value(
+                                                        &mut selected_new_description,
+                                                        Some(*child),
+                                                        name.as_str(),
+                                                    );
+                                                }
+                                            }
+                                        });
+
+                                    if let Some(selected_new_description) = selected_new_description
+                                    {
+                                        params.pending.selected = Some(selected_new_description);
+                                        add_instance = Some(
+                                            params
+                                                .next_instance_name
+                                                .get_for(selected_new_description),
+                                        );
+                                    }
+                                })
+                                .response
+                                .on_hover_text("Select Description");
+
+                                ui.add_space(5.0);
+                                ui.separator();
+
+                                // Section for creating a new description
+                                ui.label("New Model Description");
+                                ui.add_space(2.0);
+                                ui.horizontal(|ui| {
+                                    ui.label("Name");
+                                    ui.text_edit_singleline(&mut params.pending.name);
```

Comment:
What do you think about using the following approach to show background text in `Name` field? Following this by changing the default value for [`PendingModelDescription.name`](https://github.com/open-rmf/rmf_site/blob/b852877aff05bbd998f40121342e4f8dfca55af6/rmf_site_editor/src/widgets/creation.rs#L394) to an empty string.

```suggestion
                                    TextEdit::singleline(&mut params.pending.name)
                                        .hint_text("<Unnamed Description>")
                                        .show(ui);
```

Reply:
I thought this was an interesting idea so I tried it out. I found that the hint text is visually indistinguishable from normal text, e.g.:

Hint text:
![hint_text-0](https://github.com/user-attachments/assets/049c59cd-c8af-460e-b61c-39617e2a7a77)

Normal text:
![hint_text-1](https://github.com/user-attachments/assets/77715dc6-04bc-4777-bd0b-06d118be8264)

I think this is problematic because it won't be clear to the user whether they have a name typed into the field or if it's actually blank.

Another problem is the hint text doesn't show up at all unless the field is completely empty. That means there won't be an indication anywhere of what the text field is for. We could use a tooltip for that if we wanted to save screen real estate, but I think the `Name` label is compact enough that it's not detrimental.

Overall I think I'd rather stick with the current approach. Let me know what you think.

Reply:
Thanks for testing this out! That makes sense; keeping the current approach seems like the better option.